crates/steel-parser/src/tokens.rs

```diff
@@ -116,25 +116,39 @@ pub enum TokenType<S> {
     Identifier(S),
     Keyword(S),
     NumberLiteral(f64),
+    // TODO: Replace Number/Integer/Fraction literal with new enum.
+    // NumberLiteral(NumberLiteral),
     IntegerLiteral(MaybeBigInt),
     FractionLiteral(MaybeBigInt, MaybeBigInt),
     StringLiteral(String),
     Error,
 }
 
+pub enum NumberLiteral {
+    Real(RealNumberLiteral),
+    Complex(RealNumberLiteral, RealNumberLiteral),
+}
+
+pub enum RealNumberLiteral {
+    Int(MaybeBigInt),
+    Fraction(MaybeBigInt, MaybeBigInt),
+    Inexact(f64),
+}
+
 #[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize)]
 pub enum MaybeBigInt {
     Small(isize),
-    Big(BigInt),
+    Big(Box<BigInt>),
```

Comment:
Is there a specific benchmark I should measure for the token size?

Reply:
Unfortunately I don't have one specifically. In general it would probably be the engine creation one - I can link in a few minutes

Reply:
I also wouldn't be surprised if it didn't matter. I only started taking note when the size had ballooned when I wasn't paying close attention

Reply:
Ack. I may revert, especially since clippy warns when enums are getting too big. Not sure what the threshold is, but I've definitely accidentally hit it before.